python/kastore/store.py

```diff
@@ -341,8 +363,21 @@ def _read_file(self):
         # Create the mapping for descriptors.
         self._descriptor_map = {descriptor.key: descriptor for descriptor in descriptors}
 
+        if not self._read_all:
```

Comment:
I'm not a big fan of early exits like this - too easy to get confused about what happens in the different code paths. It might nicer to have a function ``_get_array(self, key)`` which had the logic for the ``read_all`` and otherwise in there.